### PR TITLE
build: add checksum check to nightly pipeline

### PR DESCRIPTION
**Pull request by rebeccabrennan**
Checksum added to Jenkins nightly file to improve security.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+set -e
```
Comment (geek-soft) [inline test.sh]: What is that needed for?

File: Jenkinsfile_nightly [hunk 2]
```diff
@@ -42,8 +42,24 @@ withNightlyPipeline(type, product, component) {
...
-    sh 'PATH=$PATH:./bin git lfs install --local && PATH=$PATH:./bin git lfs pull'
+      sh '''
+        set -e
+    
+        echo "7be85238cbbb957ab25de52b60279d40ba40d3faa72eeb2cb9fa77d6d92381e5  git-lfs-v2.7.1.tar.gz" > git-lfs-v2.7.1.checksum.txt
+        curl -L https://github.com/git-lfs/git-lfs/releases/download/v2.7.1/git-lfs-linux-amd64-v2.7.1.tar.gz > git-lfs-v2.7.1.tar.gz
+        sha256sum -c git-lfs-v2.7.1.checksum.txt
+        
+        tar xz -C bin/ git-lfs -f git-lfs-v2.7.1.tar.gz
+        ls -lash bin/
```
Comment (geek-soft) [inline Jenkinsfile_nightly]: I guess that listing is not really needed.

File: Jenkinsfile_nightly [hunk 3]
```diff
@@ -42,8 +42,24 @@ withNightlyPipeline(type, product, component) {
...
+      '''
+
+      sh '''
+        set -e
+      
+        export PATH=$PATH:./bin
+        git lfs install --local
+        git lfs pull || (git lfs logs last && false)
```
Comment (geek-soft) [inline Jenkinsfile_nightly]: Please explain what `git lfs logs last && false` does and why is that needed.

File: Jenkinsfile_nightly [hunk 4]
```diff
@@ -42,8 +42,24 @@ withNightlyPipeline(type, product, component) {
...
+    
+        echo "7be85238cbbb957ab25de52b60279d40ba40d3faa72eeb2cb9fa77d6d92381e5  git-lfs-v2.7.1.tar.gz" > git-lfs-v2.7.1.checksum.txt
+        curl -L https://github.com/git-lfs/git-lfs/releases/download/v2.7.1/git-lfs-linux-amd64-v2.7.1.tar.gz > git-lfs-v2.7.1.tar.gz
+        sha256sum -c git-lfs-v2.7.1.checksum.txt
+        
+        tar xz -C bin/ git-lfs -f git-lfs-v2.7.1.tar.gz
+        ls -lash bin/
+      '''
+
+      sh '''
```
Comment (geek-soft) [inline Jenkinsfile_nightly]: I do not see a reason to have two bash scripts. I suggest combining these together as they are both aim to achieve common goal.

File: Jenkinsfile_nightly [hunk 5]
```diff
@@ -42,8 +42,24 @@ withNightlyPipeline(type, product, component) {
...
 
   after('checkout') {
-    sh 'curl -L https://github.com/git-lfs/git-lfs/releases/download/v2.7.1/git-lfs-linux-amd64-v2.7.1.tar.gz | tar xz -C ./bin git-lfs'
-    sh 'PATH=$PATH:./bin git lfs install --local && PATH=$PATH:./bin git lfs pull'
+      sh '''
+        set -e
+    
+        echo "7be85238cbbb957ab25de52b60279d40ba40d3faa72eeb2cb9fa77d6d92381e5  git-lfs-v2.7.1.tar.gz" > git-lfs-v2.7.1.checksum.txt
+        curl -L https://github.com/git-lfs/git-lfs/releases/download/v2.7.1/git-lfs-linux-amd64-v2.7.1.tar.gz > git-lfs-v2.7.1.tar.gz
+        sha256sum -c git-lfs-v2.7.1.checksum.txt
```
Comment (geek-soft) [inline Jenkinsfile_nightly]: Can you suppress output? There might be `--status` flag to return status code only.

File: Jenkinsfile_nightly [hunk 6]
```diff
@@ -42,8 +42,24 @@ withNightlyPipeline(type, product, component) {
...
+        
+        tar xz -C bin/ git-lfs -f git-lfs-v2.7.1.tar.gz
```
Comment (geek-soft) [inline Jenkinsfile_nightly]: Please change order of arguments to make it more readable. I suggest:

```
tar xz -f git-lfs-v2.7.1.tar.gz -C bin/ git-lfs
```

This way it naturally says input file, output dir and finally output file.

File: Jenkinsfile_nightly (hunk 3 above)
Comment (rebeccabrennan) [reply Jenkinsfile_nightly]: if git lfs fails to startup it'll get the logs and then fail the build

Comment (geek-soft) [reply Jenkinsfile_nightly]: So how does it help? If we bring some code we should know what it does, am I right?

Comment (geek-soft) [reply Jenkinsfile_nightly]: Also I am interested to know what was the initial issue with git-lfs. Can you explain me why it used to fail?

File: Jenkinsfile_nightly (hunk 4 above)
Comment (rebeccabrennan) [reply Jenkinsfile_nightly]: 👍

File: Jenkinsfile_nightly (hunk 6 above)
Comment (rebeccabrennan) [reply Jenkinsfile_nightly]: 👍

File: Jenkinsfile_nightly (hunk 3 above)
Comment (rebeccabrennan) [reply Jenkinsfile_nightly]: it did not like that it was in the build folder removing it from there made it work

Comment (rebeccabrennan) [reply Jenkinsfile_nightly]: 👍

File: Jenkinsfile_nightly (hunk 2 above)
Comment (rebeccabrennan) [reply Jenkinsfile_nightly]: 👍

File: test.sh (hunk 1 above)
Comment (rebeccabrennan) [reply test.sh]: set e means fail on errors. I was taking security precautions I felt may be helpful. Although removing the test.sh script as we do not need it anymore do we?

File: Jenkinsfile_nightly (hunk 5 above)
Comment (rebeccabrennan) [reply Jenkinsfile_nightly]: we could try something like https://unix.stackexchange.com/questions/119648/redirecting-to-dev-null/119650#119650 do you think this would be helpful though there is only minimal output

File: test.sh (hunk 1 above)
Comment (geek-soft) [reply test.sh]: I meant whole file.

File: Jenkinsfile_nightly (hunk 5 above)
Comment (geek-soft) [reply Jenkinsfile_nightly]: Have you tried `--status` flag?

File: test.sh (hunk 1 above)
Comment (rebeccabrennan) [reply test.sh]: yeah removed

File: Jenkinsfile_nightly (hunk 5 above)
Comment (rebeccabrennan) [reply Jenkinsfile_nightly]: where do you want the --status flag ?
